### PR TITLE
wasmplugin: add prefix to resource name

### PR DESCRIPTION
**Pull request by zirain**
add prefix to the resource name of WasmPlugin, preventing the name collision with other resources.

### CODE REVIEW

**Comment by keithmattix** [comment]
What's the context? Why do we need the prefix?

**Comment by zirain** [comment]
> What's the context? Why do we need the prefix?

trying to extend the usage of ECDS, add a prefix to differ from others.

**Comment by keithmattix** [comment]
Ah I see to prevent conflicts between WASMPlugin and other ECDS resources like Telemetry API. Might be worth an ecds architecture doc in the future, but for now, could you just add comment on the ResourceName field indicating that a prefix should always be used? Also consider a test that asserts WASMPlugins are prefixed

File: pilot/pkg/model/extensions.go [hunk 1]
```diff
@@ -49,6 +49,10 @@ const (
 	WasmPolicyEnv = "ISTIO_META_WASM_IMAGE_PULL_POLICY"
 	// name of environment variable at Wasm VM, which will carry the resource version of WasmPlugin.
 	WasmResourceVersionEnv = "ISTIO_META_WASM_PLUGIN_RESOURCE_VERSION"
+
+	// WasmPluginResourceNamePrefix is the prefix of the resource name of WasmPlugin,
+	// preventing the name collision with other resources.
+	WasmPluginResourceNamePrefix = "istio.io/wasmplugin/"
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/extensions.go:55]: if we use `group/kind/namespace/name` it should be `extenstions.istio.io/Wasmplugin/`

**Review by hzxuzhonghu — COMMENTED** [review]
Add the ctx in pr description

File: pilot/pkg/model/extensions.go (hunk 1 above)
Comment (zirain) [reply pilot/pkg/model/extensions.go:55]: sounds good

**Review by hzxuzhonghu — APPROVED** [review]
LGTM

**Pull request merged**